Sources/MongoSwift/MongoClient.swift

```diff
@@ -187,9 +187,14 @@ public class MongoClient {
             throw UserError.invalidArgumentError(message: "libmongoc not built with TLS support")
         }
 
+        self.encoder = BSONEncoder(options: options)
+        self.decoder = BSONDecoder(options: options)
+
         // if a readConcern is provided, set it on the client
         if let rc = options?.readConcern {
-            mongoc_client_set_read_concern(self._client, rc._readConcern)
+            try? rc.withMongocReadConcern { tmpReadConcernPtr in
```

Comment:
I don't think we want to `try?` here. if there is an error, we should propagate it to the user using `try`.

Reply:
I think my comment below about using `rethrows` will make this comment obsolete ^